lib/active_shipping/carriers/canada_post_pws.rb

```diff
@@ -158,13 +158,14 @@ def maximum_weight
     # service discovery
 
     def parse_services_response(response)
-      doc = REXML::Document.new(response)
-      service_nodes = doc.elements['services'].elements.collect('service') { |node| node }
+      doc = Nokogiri.XML(response)
+      doc.remove_namespaces!
+      service_nodes = doc.xpath('services/service')
       service_nodes.inject({}) do |result, node|
-        service_code = node.get_text("service-code").to_s
-        service_name = node.get_text("service-name").to_s
-        service_link = node.elements["link"].attributes['href']
-        service_link_media_type = node.elements["link"].attributes['media-type']
+        service_code = node.at("service-code").text
+        service_name = node.at("service-name").text
+        service_link = node.at("link").attributes['href'].value
```

Comment:
Yeah, it will raise an exception in that case. I can fix it so it returns `nil` instead of blowing up, but I find it highly unlikely that the `<link>` tag is there without an `href` attribute.